python/detail.h

```diff
@@ -14,3 +14,9 @@ struct MoveableVariable {
 struct MoveableDataArray {
   dataset::DataArray data;
 };
+
+template <class T> using CstViewRef = const typename T::const_view_type &;
+
+template <class T> using ViewRef = const typename T::view_type &;
+
+template <class T> using View = typename T::view_type;
```

Comment:
Not sure about these helpers, they require more memorization of variable/trait names? If anything, go with something more standard: `view_t` instead of `View` (and `const_view_t`?)?

I would recommend to not have the "Ref" versions though, spelling the `const` and `&` explicitly where things are called is much cleared.

Reply:
Yes, i agree about spelling the `const` and `&`. Will change

Reply:
So these were actually inspired from https://github.com/scipp/scipp/blob/master/python/operations.cpp#L18

So you think it would be better if I had
```
  using const_view_t = typename T::const_view_type;
  m.def("flatten", py::overload_cast<const const_view_t &, const Dim>(&flatten),
```
instead of
```
  using ConstView = const typename T::const_view_type &;
  m.def("flatten", py::overload_cast<ConstView, const Dim>(&flatten),
```
?

Reply:
No, the point was not about local typedefs within a function (I am fine with any name then). But *if* you add a generic helper trait for this, it should resemble what the standard library does, e.g., 

```cpp
template< bool B, class T, class F >
using conditional_t = typename conditional<B,T,F>::type;
```
... but actually our case is slightly different, so I am not sure.